app/domain/authentication/authn_jwt/jwt_configuration_factory.rb

```diff
@@ -0,0 +1,17 @@
+module Authentication
+  module AuthnJwt
+    # Factory that receives a vendor name and returns the appropriate JWT vendor configuration class
+    class JwtConfigurationFactory
+      VENDORS = {
+        "dummy" => JWTConfigurationDummyVendor
+      }
+
+      def create_jwt_configuration(vendor)
+        unless VENDORS.key?(vendor)
```

Comment:
Instead of looking of the key twice, you can do it once and raise if you don't find value.